tools/check-links

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python3
+import argparse
+import re
+import sys
+import threading
+import urllib.request
+from urllib.error import HTTPError
+from urllib.parse import urldefrag, urljoin, urlparse
+from urllib.request import Request, urlopen
+
+import requests
+from bs4 import BeautifulSoup
+
+allow_mails = ["strager.nds@gmail.com"]
+
+_allowed_file_ext_for_soup_ = [
+    "html",
+    "js",
+]
+
+headers = {
+    "User-Agent": "Mozilla/5.0 (Windows NT 6.1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/41.0.2228.0 Safari/537.3"
+}
+
+
+def check_fragment(soup, fragment) -> bool:
+    if fragment:
+        return True if soup.select(f"#{fragment}") else False
+    return True
+
+
+def is_mailto_link(link) -> bool:
+    return True if re.search(r"^mailto:*", link) else False
+
+
+def get_mail_address(link) -> str:
+    return link.split(":")[1]
+
+
+def worker(parent_url, url):
+    try:
+        req = urllib.request.Request(url, headers=headers)
+        response = urlopen(req)
+    except HTTPError as e:
+        print(f"({e.code} error) {parent_url}, {url}")
+
+
+class Crawler:
+    def __init__(self, site, allowed_file_ext_for_soup=_allowed_file_ext_for_soup_):
+        self.site = site
+        self.allowed_file_ext_for_soup = allowed_file_ext_for_soup
+        self.site_namespace = urlparse(self.site).netloc
+        self.visted_urls = []
+        self.visted_urls_soup = {}
+        self.threads = list()
+
+        try:
+            result = requests.get(self.site)
+            soup = BeautifulSoup(result.text, "html.parser")
+            self.urls = self.get_urls_from_page(soup)
+        except requests.exceptions.ConnectionError:
+            print("(error) failed to get", self.site)
+
+    def in_namespace(self, url) -> bool:
+        return urlparse(url).netloc == self.site_namespace
+
+    def get_urls_from_page(self, soup) -> list:
+        href_tags = soup.find_all(href=True)
+        hrefs = [tag.get("href") for tag in href_tags]
+        return hrefs
```

Comment:
We should also crawl scripts referenced by `<script src=...>`. See website/benchmarks/index.html for an example.